lucille-core/src/main/java/com/kmwllc/lucille/indexer/ElasticsearchIndexer.java

```diff
@@ -91,24 +95,27 @@ protected void sendToIndex(List<Document> documents) throws Exception {
       // handle special operations required to add children documents
       addChildren(doc, indexerDoc);
 
+      // we don't have to worry about routing being null or not in the document since the indexer will ignore it
+      String routing = doc.getString(routingField);
+
       if (update) {
         br.operations(op -> op
             .update(up -> up
                 .id(docId)
                 .index(index)
+                .routing(routing)
```

Comment:
abi has been working on the equivalent code in OpenSearchIndexer
we couldn't find documentation saying that .routing(null) would be ignored, so we decided to be on the safe side and only set the routing when non-null
along with routing, there is also version info that we should set if versioning is configured
ideally the ElasticsearchIndexer and OpenSearchIndexers should be kept as close as possible, so let's compare and reconcile
here is the current code in abi's PR: https://github.com/kmwtechnology/lucille/blob/9bff23eeafa78c00431b747fd3d83b97a864ca01/lucille-core/src/main/java/com/kmwllc/lucille/indexer/OpenSearchIndexer.java#L124